net/api/client/client_test.go

```diff
@@ -202,11 +202,16 @@ func TestClient_AddRecord(t *testing.T) {
 	// Create a thread, keeping read key and log private key on the client
 	id := thread.NewIDV1(thread.Raw, 32)
 	tk := thread.NewRandomServiceKey()
-	sk, pk, err := crypto.GenerateEd25519Key(crand.Reader)
+	logSk, logPk, err := crypto.GenerateEd25519Key(crand.Reader)
 	if err != nil {
 		t.Fatal(err)
 	}
-	info, err := client.CreateThread(context.Background(), id, core.ThreadKey(tk), core.LogKey(pk))
+	authorSk, _, err := crypto.GenerateEd25519Key(crand.Reader)
+	if err != nil {
+		t.Fatal(err)
+	}
+	creds := thread.NewDefaultCreds(id, thread.WithPrivKey(authorSk))
+	_, err = client.CreateThread(context.Background(), creds, core.WithThreadKey(tk), core.WithLogKey(logPk))
```

Comment:
Here's an example of creating credentials that actually have an author. The default thread creds constructor can take an optional private key.